internal/daemonset/daemonset.go

```
@@ -0,0 +1,199 @@
+			{
+				MountPath: "/host",
+				Name:      "host",
+			},
+		},
+	}
+	return &v1.DaemonSet{
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        dsName,
+			Namespace:   namespace,
+			Annotations: dsAnnotations,
+		},
+		Spec: v1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: matchLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers:       []corev1.Container{container},
+					PreemptionPolicy: &preempt,
+					Priority:         &zeroInt32,
+					HostNetwork:      true,
+					Tolerations: []corev1.Toleration{
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/not-ready",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/unreachable",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect: "NoSchedule",
+							Key:    "node-role.kubernetes.io/master",
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+									Type: &hostPathType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Delete daemon set
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	tnf.ClaimFilePrintf("Deleting daemon set %s\n", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	if err := client.DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	}); err != nil {
+		tnf.ClaimFilePrintf("The daemonset (%s) deletion is unsuccessful due to %s", daemonSetName, err)
```

You should return with the error right here, as it doesn't make any sense to continue after the trace. You can just return a custom error like this:
```
return fmt.Errorf("daemonset %s deletion failed: %w", daemonSetName, err)
```

---

I agree.